cli.ts

```diff
@@ -59,18 +59,21 @@ if (core.getBooleanInput("preprocess")) {
     console.log("Moved README.md to Home.md");
   }
 
-  const mdRe = /\.(?:md|markdown|mdown|mkdn|mkd|mdwn|mkdown|ron)$/;
+  const mdRe = /\.(?:md|markdown|mdown|mkdn|mkd|mdwn|mkdown|ron)(#[\w\-]*)?$/;
   const plugin = () => (tree: any) =>
     visit(tree, ["link", "linkReference"], (node: any) => {
-      if (!mdRe.test(node.url)) {
+      const matches = node.url?.match(mdRe)
+      if (matches === null || matches === undefined) {
         return;
       }
       if (!new URL(node.url, "file:///-/").href.startsWith("file:///-/")) {
         return;
       }
 
       const x = node.url;
-      node.url = node.url.replace(mdRe, "");
+      node.url = matches.length === 2
+        ? node.url.replace(mdRe, "$1") 
```

Comment:
TODO in the future: GitHub wikis can have custom sidebars enabling nested subdirectories. This may break those cases; I think it's fine to break them for now since these links didn't work before anyways

Reply:
you're right. these seem to be broken. my test wiki (in this PR at the moment) has a document in a subdirectory. the generated structure flattens it, so that `wiki/somewhere-else/another.page.md` becomes just [/wiki/another-page](https://github.com/chinatsu/github-wiki-action/wiki/another-page). i'm not sure if i did something wrong here.

Reply:
i believe you may have misunderstood what custom sidebars achieve. https://github.com/orgs/community/discussions/23914

they are purely a way to visually make it seem like the wiki structure contains nested subdirectories. 
the wiki in actuality does _not_ support directories, as you may see from the lamented comments as recent as this year on this topic posted nearly 7 years ago.